upstairs/src/lib.rs

```diff
@@ -5685,10 +5689,7 @@ impl Upstairs {
          * Determine the highest flush number and make sure our generation
          * is high enough.
          */
-        let need_repair;
-        let repair_commands;
-        let failed_collate;
-        {
+        let collate_status = {
```

Comment:
This part is the meat of the change, instead of having these three booleans, we just
have one `collate_status` now, that is a `Result<bool, CrucibleError>`.  This was what the
three variables were trying to express, and now it's all wrapped up nicely in a single place.

Reply:
agreed that this is much nicer!